src/main-client.ts

```diff
@@ -1219,234 +1890,1383 @@ export class MainClient extends BaseRestClient {
+  getFlexibleRedemptionRecord(
+    params: GetFlexibleRedemptionRecordParams,
+  ): Promise<{
+    rows: FlexibleRedemptionRecord[];
+    total: number;
+  }> {
+    return this.getPrivate(
+      'sapi/v1/simple-earn/flexible/history/redemptionRecord',
+      params,
+    );
   }
 
-  getStakingProductPosition(
-    params: StakingBasicParams & {
-      productId?: string;
-      asset?: string;
-    },
-  ): Promise<StakingProductPosition[]> {
-    return this.getPrivate('sapi/v1/staking/position', params);
+  getLockedRedemptionRecord(params: GetLockedRedemptionRecordParams): Promise<{
+    rows: LockedRedemptionRecord[];
+    total: number;
+  }> {
+    return this.getPrivate(
+      'sapi/v1/simple-earn/locked/history/redemptionRecord',
+      params,
+    );
   }
 
-  getStakingHistory(params: StakingHistoryParams): Promise<StakingHistory[]> {
-    return this.getPrivate('sapi/v1/staking/stakingRecord', params);
+  getFlexibleRewardsHistory(params: GetFlexibleRewardsHistoryParams): Promise<{
+    rows: FlexibleRewardsHistory[];
+    total: number;
+  }> {
+    return this.getPrivate(
+      'sapi/v1/simple-earn/flexible/history/rewardsRecord',
+      params,
+    );
   }
 
+  getLockedRewardsHistory(params: GetLockedRewardsHistoryParams): Promise<{
+    rows: GetLockedRewardsHistory[];
+    total: number;
+  }> {
+    return this.getPrivate(
+      'sapi/v1/simple-earn/locked/history/rewardsRecord',
+      params,
+    );
+  }
+
+  setFlexibleAutoSubscribe(params: SetAutoSubscribeParams): Promise<{
+    success: boolean;
+  }> {
+    return this.postPrivate(
+      'sapi/v1/simple-earn/flexible/setAutoSubscribe',
+      params,
+    );
+  }
+
+  setLockedAutoSubscribe(params: SetAutoSubscribeParams): Promise<{
+    success: boolean;
+  }> {
+    return this.postPrivate(
+      'sapi/v1/simple-earn/locked/setAutoSubscribe',
+      params,
+    );
+  }
+
+  getFlexiblePersonalLeftQuota(params: { productId: string }): Promise<{
+    leftPersonalQuota: string;
+  }> {
+    return this.getPrivate(
+      'sapi/v1/simple-earn/flexible/personalLeftQuota',
+      params,
+    );
+  }
+
+  getLockedPersonalLeftQuota(params: { projectId: string }): Promise<{
+    leftPersonalQuota: string;
+  }> {
+    return this.getPrivate(
+      'sapi/v1/simple-earn/locked/personalLeftQuota',
+      params,
+    );
+  }
+
+  getFlexibleSubscriptionPreview(
+    params: GetFlexibleSubscriptionPreviewParams,
+  ): Promise<GetFlexibleSubscriptionPreviewResponse> {
+    return this.getPrivate(
+      'sapi/v1/simple-earn/flexible/subscriptionPreview',
+      params,
+    );
+  }
+
+  getLockedSubscriptionPreview(
+    params: GetLockedSubscriptionPreviewParams,
+  ): Promise<GetLockedSubscriptionPreviewResponse[]> {
+    return this.getPrivate(
+      'sapi/v1/simple-earn/locked/subscriptionPreview',
+      params,
+    );
+  }
+
+  getRateHistory(params: GetRateHistoryParams): Promise<{
+    rows: GetRateHistory[];
```

Comment:
```suggestion
    rows: RateHistory[];
```

Call it what it is, each row is a "RateHistory" (rename the interface too pls)